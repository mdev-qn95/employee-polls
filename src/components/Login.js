const Login = () => {
  return (
    <div>
      <span>LOGIN</span>
    </div>
  );
};

export default Login;
