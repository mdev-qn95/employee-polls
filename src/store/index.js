import { configureStore } from "@reduxjs/toolkit";
import users from "./reducers/usersReducers";
import authed from "./reducers/authedReducers";
import questions from "./reducers/questionsReducers";

export const store = configureStore({
  reducer: {
    authed,
    users,
    questions,
  },
});
