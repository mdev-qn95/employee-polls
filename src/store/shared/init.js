import { getUserInformation } from "../../util/api";
import { getQuestions } from "../actions/questionsAction";
import { getUser } from "../actions/usersAction";

export function getDataInit() {
  return (dispatch) => {
    return getUserInformation().then(({ users, questions }) => {
      dispatch(getUser(users));
      dispatch(getQuestions(questions));
    });
  };
}
