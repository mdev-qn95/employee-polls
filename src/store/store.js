import { configureStore } from "@reduxjs/toolkit";
import authed from "./reducers/authedReducers";
import users from "./reducers/usersReducers";
import questions from "./reducers/questionsReducers";

export const store = configureStore({
  reducer: {
    authed,
    users,
    questions,
  },
});
