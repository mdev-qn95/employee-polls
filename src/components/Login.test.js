import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import Login from "./Login";
import { store } from "../store";
import { getDataInit } from "../store/shared/init";

describe("Login", () => {
  it("will render the component", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("will clear the value of the inputs after pressing the submit button", async () => {
    await store.dispatch(getDataInit());

    const wrapper = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );

    const loginPage = wrapper.getByTestId("login-page");
    const userIdEle = wrapper.getByTestId("user-id");
    const passwordEle = wrapper.getByTestId("password");
    const submitEle = wrapper.getByTestId("submit");

    expect(loginPage).toBeInTheDocument();
    expect(userIdEle).toBeInTheDocument();
    expect(passwordEle).toBeInTheDocument();
    expect(submitEle).toBeInTheDocument();

    fireEvent.change(userIdEle, { target: { value: "johndoe" } });
    fireEvent.change(passwordEle, {
      target: { value: "000" },
    });

    expect(userIdEle.value).toBe("johndoe");
    expect(passwordEle.value).toBe("000");

    fireEvent.click(submitEle);

    expect(loginPage).toBeInTheDocument();
    expect(userIdEle.value).toBe("");
    expect(passwordEle.value).toBe("");
  });
});
