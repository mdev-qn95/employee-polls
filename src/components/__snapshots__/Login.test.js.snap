// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login will render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-2 container"
        data-testid="login-page"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h2>
              Employee Polls
            </h2>
            <img
              alt="Login"
              class=""
              src="login.png"
              width="350px"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h2>
              Log In
            </h2>
            <form
              class=""
            >
              <div
                class="mt-2"
              >
                <label
                  class="form-label"
                  for="user"
                >
                  User
                </label>
                <input
                  class="form-control"
                  data-testid="user-id"
                  id="user"
                  name="user"
                  placeholder="User"
                  type="text"
                  value="johndoe"
                />
              </div>
              <div
                class="mt-2"
              >
                <label
                  class="form-label"
                  for="password"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  data-testid="password"
                  id="password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value="johndoe"
                />
              </div>
              <button
                class="mt-5 btn btn-primary"
                data-testid="submit"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-2 container"
      data-testid="login-page"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h2>
            Employee Polls
          </h2>
          <img
            alt="Login"
            class=""
            src="login.png"
            width="350px"
          />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h2>
            Log In
          </h2>
          <form
            class=""
          >
            <div
              class="mt-2"
            >
              <label
                class="form-label"
                for="user"
              >
                User
              </label>
              <input
                class="form-control"
                data-testid="user-id"
                id="user"
                name="user"
                placeholder="User"
                type="text"
                value="johndoe"
              />
            </div>
            <div
              class="mt-2"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                data-testid="password"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value="johndoe"
              />
            </div>
            <button
              class="mt-5 btn btn-primary"
              data-testid="submit"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
